app/models/form526_submission.rb

```diff
@@ -6,10 +6,82 @@
 
 class Form526Submission < ApplicationRecord
   extend Logging::ThirdPartyTransaction::MethodWrapper
-
+  include AASM
   include SentryLogging
   include Form526ClaimFastTrackingConcern
 
+  # Documentation:
+  # https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/products/disability/526ez/implementation/form_526_state_machine.md
+  aasm do
+    after_all_transitions :log_status_change
+
+    state :unprocessed, initial: true
+    state :delivered_to_primary, :failed_primary_delivery, :rejected_by_primary,
+          :delivered_to_backup, :failed_backup_delivery, :rejected_by_backup,
+          :in_remediation, :finalized_as_successful, :unprocessable
+
+    # - a submission has been delivered to our happy path
+    # - requires polling to finalize
+    event :deliver_to_primary do
+      transitions to: :delivered_to_primary
+    end
+
+    # - submission failed delivery to primary path for any reason
+    # - requires backup submission or remediation
+    event :fail_primary_delivery do
+      transitions to: :failed_primary_delivery
+    end
+
+    # - a successfully delivered submission has failed 3rd party validations on primary path
+    # - requires backup submission or remediation
+    event :reject_from_primary do
+      transitions to: :rejected_by_primary
+    end
+
+    # - a submission has been delivered to our backup path
+    # - accepts any non-success type state
+    # - MUST accept all the same incoming statuses as fail_backup_delivery
```

Comment:
This is the same comment as on L51. Will you explain why the `deliver_to_backup` event MUST accept all the same incoming statuses as `fail_backup_delivery`?

Reply:
Those comments are outdated; removing them.  check the docs if you want to better grok to actual lifecycle flows that will (probalby) happen.  Again, we are allowing uncertainty by not enforcing `from`  state because the primary value here is dog-earing remediated submissions.